lib/ansiblelint/rules/ComparisonToEmptyStringRule.py

```diff
@@ -19,5 +23,5 @@ class ComparisonToEmptyStringRule(AnsibleLintRule):
 
     empty_string_compare = re.compile("[=!]= ?(\"{2}|'{2})")
 
-    def match(self, file, line):
-        return self.empty_string_compare.search(line)
+    def match(self, file: "TargetFile", line: str = "") -> Union[bool, str]:
```

Comment:
I think per our earlier discussion, the return value should be bool and any strings would need to be attached to an exception raised. Or alternatively, the return type should be a specialized object.

Reply:
I was considering returning `List[MatchError]` but this may count as a potential breaking API change.

Using this makes it future proof by allowing to return multiple matches with different messages.

The breakage risk is low because we happen to control the likely single place where this is used, matchlines() from based class. 

@webknjaz What do you think? -- List[MatchError] is already the return type of matchlines().

Reply:
Why list? Do you mean that it'd return the same violation on multiple lines? Why not implement a list in the `MatchError` then?

Reply:
Errors can be at different lines and if you check which attributes MatchError has you will see that it cannot be used to host multiple messages or multiple lines.

Some rules are more dynamic with their errors, for example galaxy one can identify multiple issues.